tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -631,4 +649,28 @@ function() {
 		$tag = wp_get_attachment_image( $attachment_id, 'medium', false, array( 'class' => "wp-image-{$attachment_id}" ) );
 		$this->assertNotSame( $tag, webp_uploads_img_tag_update_mime_type( $tag, 'the_content', $attachment_id ) );
 	}
+
+	/**
+	 * Tests whether additional mime types generated only for allowed image sizes or not.
+	 *
+	 * @test
+	 */
+	public function it_should_create_mime_types_for_allowed_sizes_only() {
+		remove_all_filters( 'webp_uploads_image_sizes_with_additional_mime_type_support' );
+		add_filter(
+			'webp_uploads_image_sizes_with_additional_mime_type_support',
+			function( $sizes ) {
+				$sizes['allowed_size_400x300'] = 'allowed_size_400x300';
+				return $sizes;
+			}
+		);
+
+		add_image_size( 'allowed_size_400x300', 400, 300, true );
+		add_image_size( 'not_allowed_size_200x150', 200, 150, true );
+
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/car.jpeg' );
+
```

Comment:
```suggestion

	// Clean up.
	remove_image_size( 'allowed_size_400x300' );
	remove_image_size( 'not_allowed_size_200x150' );
```

The test suite doesn't clean up image sizes between tests, so best to so so manually to avoid subsequent tests generating these image sizes.

Reply:
Thanks. Updated.